lib/galaxy/tool_util/toolbox/base.py

```diff
@@ -408,6 +408,12 @@ def load_item(
                 panel_dict = self._tool_panel
             if integrated_panel_dict is None:
                 integrated_panel_dict = self._integrated_tool_panel
+            load_if = item.get("if")
+            if load_if == "interactivetools_enable":
```

Comment:
This seems to duplicate in the tool conf the info provided by `tool_type="interactive"` in the tool, not sure I like it.

Reply:
I get why you feel that way but I disagree. I feel like the toolbox says this is what should be loaded and it should be an error if you are attempting to load something that isn't configured properly. It only feels like duplicated information because we aren't serious enough about tool errors - we just ignore stuff and keep going with loading the tool panel.

Reply:
I don't think you would suggest Galaxy should not start if a tool doesn't load correctly, so "just ignore stuff and keep going with loading the tool panel" seems the right approach to me.
We could present the tool load failure errors in the admin UI instead of keeping them in the logs, but that's another discussion.

Reply:
It is what we do for every other configuration error - it is what an application should do if it is misconfigured IMO. I'm closing this.

Reply:
Yes, but these are user plugins .. say ITs are on the tool shed one day (nothing is actually stopping a "rogue" actor from uploading ITs). If an admin installs a tool and restarts the server it won't restart. That is a heavy price to pay for forcing that the application config is correct.